README.md

```diff
@@ -2124,10 +2124,95 @@ the build will fail if test coverage is below 100%.
 
 You should be able to do the typical stuff:
 
+### 1. Cloning repository:
+Fork the repository and run this(Only for SSH):
+
+```
+git clone git@github.com:${your_github_username}/foobara.git
+```
+
+Create a new branch for you to push into
+
+```
+git checkout -b <branch-name>
+```
+
+Now navigate to project directory
+
 ```
-git clone git@github.com:foobara/foobara
 cd foobara
+```
+
+### 2. Installing mise (LINUX/WSL)
+Mise is a package manager and helps with managing different versions of ruby. It allows you to switch different versions of ruby.
+
+The installation docs might be updated by mise so here's the reference to that: [Installation Docs](https://mise.jdx.dev/installing-mise.html)
+
+***If you already installed by following mise's docs then, you can skip this section***
+
+Else to setup mise follow the script given below:
+
+```
+sudo apt update -y && sudo apt install -y gpg wget \
+build-essential \
+  libssl-dev zlib1g-dev libreadline-dev libyaml-dev libxml2-dev \
+  libxslt1-dev libffi-dev libgdbm-dev autoconf bison
+wget -qO - https://mise.jdx.dev/gpg-key.pub | gpg --dearmor | sudo tee /etc/apt/keyrings/mise-archive-keyring.gpg 1> /dev/null
+echo "deb [signed-by=/etc/apt/keyrings/mise-archive-keyring.gpg arch=amd64] https://mise.jdx.dev/deb stable main" | sudo tee /etc/apt/sources.list.d/mise.list
+sudo apt update
+sudo apt install -y mise
+```
+
+Verify if it's installed or not by running:
+
+```
+mise -v
+```
+
+It should print out an ascii-art saying "mise-en-place"
+
+### 3. Install required ruby version
+To automatically install the required version mentioned in the .ruby-version in the project's directory, run this command
```

Comment:
I could test this again to be sure, but I'm pretty sure that this only works if you already have a ruby version installed. So we might need to bring back the use 3.4 line that was here before.